lib/trie: add lua bindings

tomaskrizek opened the pull request:
Trie can be useful as a general purpose high-performance data structure, especially when dealing with large amounts of IP address. Since the C code is already there, I just moved it around and added bindings so we can use it from Lua.

Depends on:
- #147 
- #149

jelu commented:
@tomaskrizek Can you make a pass on the documentation for the newline man-page thingy and add a ref to this new library in `src/lib.lua`?

tomaskrizek commented:
Fixed the doc and rebased.

I also added a WIP commit that removes the size check in `ip.tostring()`. I didn't realize I actually have `ctype<const unsigned char *>` which I want to convert to string representation. I currently don't see a way to keep the size check, but there might be some solution.

tomaskrizek commented:
> I also added a WIP commit that removes the size check in `ip.tostring()`. I didn't realize I actually have `ctype<const unsigned char *>` which I want to convert to string representation. I currently don't see a way to keep the size check, but there might be some solution.

Nevermind this comment, I've fixed this in https://github.com/DNS-OARC/dnsjit/pull/150/commits/fb4e36e16b1e1f687798501e90af57567dc455ea
